src/elm/Pages/Build/Model.elm

```diff
@@ -32,34 +37,72 @@ import Vela
 type alias PartialModel a =
     { a
         | navigationKey : Navigation.Key
+        , sourceRepos : WebData SourceRepositories
+        , user : WebData CurrentUser
+        , page : Page
         , time : Posix
         , zone : Zone
         , repo : RepoModel
         , shift : Bool
+        , pipeline : PipelineModel
     }
 
 
 
 -- TYPES
 
 
-type Msg
-    = ExpandStep Org Repo BuildNumber StepNumber
-    | FocusLogs String
-    | DownloadLogs String String
-    | FollowStep Int
-    | ExpandAllSteps Org Repo BuildNumber
-    | CollapseAllSteps
-    | FocusOn String
+type alias Msgs msg =
+    { collapseAllSteps : msg
+    , expandAllSteps : ExpandAll msg
+    , expandStep : Expand msg
+    , collapseAllServices : msg
+    , expandAllServices : ExpandAll msg
+    , expandService : Expand msg
+    , logsMsgs : LogsMsgs msg
+    }
```

Comment:
now we can add services to our component msg bus
this is where moving Msg back to Main.elm is becoming extremely useful..... trial by fire 🔥